app/filtering.js

```diff
@@ -640,13 +641,81 @@ function registerForMagnetHandler (session) {
   }
 }
 
+/**
+ * Checks that a Tor proxy is available on socks5 port 9050
+ */
+module.exports.checkTorAvailable = () => {
+  const ses = registeredSessions['tor-test']
+  if (!ses) {
+    initPartition('tor-test')
+    module.exports.checkTorAvailable()
+    return
+  }
+
+  const proxyConfig = {
+    proxyRules: 'socks5://127.0.0.1:9050,direct://'
+  }
+  // XXX: due to muon bug, callback is required but doesn't do anything
+  ses.setProxy(proxyConfig, () => {})
+
+  // TODO: remove setTimeout once muon supports setProxy callback
+  setTimeout(() => {
+    request('https://check.torproject.org/?TorButton=true', (err, response, body) => {
+      let success = null
+      if (err) {
+        console.log('Could not check Tor status', err)
+      } else if (body && body.includes('id="TorCheckResult" target="success"')) {
+        // success
+        console.log('Tor is running!')
+        success = true
+      } else if (body && body.includes('id="TorCheckResult" target="failure"')) {
+        // we are not using Tor
+        console.log('Tor is not running.')
+        success = false
+      } else {
+        // unknown error
+        console.log('Bad response from check.torproject.org. HTTP status:',
+          response.statusCode)
+      }
+      if (!success) {
+        const message =
+          locale.translation(success === false ? 'torCheckFailure' : 'torCheckError')
+        appActions.showNotification({
+          position: 'global',
+          message,
+          buttons: [
+            {text: locale.translation('dismiss')}
+          ],
+          options: {
+            persist: false
+          }
+        })
+        permissionCallbacks[message] = () => {
+          appActions.hideNotification(message)
+        }
+      }
+    }, ses)
+  }, 500)
+}
+
+module.exports.setTorNewIdentity = (url) => {
+  const origin = getOrigin(url)
+  const ses = session.fromPartition('persist:tor')
+  if (!ses || !origin) {
+    return
+  }
+  ses.setTorNewIdentity(origin)
```

Comment:
it seems easier to just do `ses.setTorNewIdentity(url)` here instead of `ses.setTorNewIdentity(origin)` since new identity is actually set by host in Muon, not origin:  https://github.com/brave/muon/pull/473/commits/7e052dd064e307a267e51202d6b7fab5b694c7fd#diff-df5c427fd693f9ebb2e5952cf0e9c8c7R744

Reply:
just fixed in 9048d8a5ac1e4ebe96027306b0fc087051a9aae1

Reply:
btw, the correct value should be `SiteInstance`
https://github.com/brave/muon/pull/473/commits/c3d6769ba628f4da20d9907dc44dfb6821036d03#diff-df5c427fd693f9ebb2e5952cf0e9c8c7R749